### DISCUSSION TITLE
Clockwork integration

### DISCUSSION THREAD

**Discussion by guilhermeaiolfi** [cycle/orm#497] [Q&A]
I was wondering if anyone attempt to create a CycleDataSource for clockwork (https://github.com/itsgoingd/clockwork)?

If not, what would be de best way to apporach it? Because I'm willing to try that.

**Comment by roxblnfk** [comment]
Hi 🖐️

I'm not sure but I think there is no an integration Cycle with Clockwork.

If Clockwork supports PSR logger, you can use it: configure the logger for Database driver and set the option `logQueryParameters = true`. It will send SQL query, parameters, and execution time.
There is also the [`logInterpolatedQueries`](https://github.com/cycle/database/pull/165) option, but it's not covered by the documentation.
https://cycle-orm.dev/docs/database-connect/current/en#additional-connection-options
